pkg/scheduler/algorithm/priorities/requested_to_capacity_ratio.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package priorities
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"
+	"k8s.io/kubernetes/pkg/scheduler/schedulercache"
+)
+
+// FunctionShape represents shape of scoring function.
+// For safety use NewFunctionShape which performs precondition checks for struct creation.
+type FunctionShape struct {
```

Comment:
Do you still need this struct? Why not just using an array of `UtilizationShapePoint`s (`[]UtilizationShapePoint`)?

Reply:
We could. I did not wanted to have separation between config API and internal implementation. Hence separate structs.

Reply:
Even if you want a separate struct, it would be better to keep them in sync. I mean make this an array of (int, int) similar to a UtilizationShapePoint.